src/mscorlib/corefx/System/Globalization/TextInfo.Unix.cs

```diff
@@ -61,9 +60,7 @@ private unsafe char ChangeCase(char c, bool toUpper)
         private bool NeedsTurkishCasing(string localeName)
         {
             Contract.Assert(localeName != null);
-
-            string lcName = CultureData.AnsiToLower(localeName);
-            return lcName.Length >= 2 && ((lcName[0] == 't' && lcName[1] == 'r') || (lcName[0] == 'a' && lcName[1] == 'z'));
+            return CultureInfo.GetCultureInfo(localeName).CompareInfo.Compare("i", "I", CompareOptions.IgnoreCase) != 0;
```

Comment:
I can't tell... is this a relatively cheap operation? (Looks like it could be hefty, but maybe lots of things get cached.)  Or are TextInfo's constructed fairly rarely?